hplsql/src/test/queries/local/drop_package.sql

```diff
@@ -0,0 +1,21 @@
+CREATE PACKAGE Counter AS
+  count INT := 0;
+  FUNCTION current() RETURNS INT;
+  PROCEDURE inc(i INT);
+END;
+
+CREATE PACKAGE BODY Counter AS
+  FUNCTION current() RETURNS INT IS BEGIN RETURN count; END;
+  PROCEDURE inc(i INT) IS BEGIN count := count + i; END;
+END;
```

Comment:
I think the language should be changed - we have separate `create package` and `create pacakge body`

and replace those 2 with a single `create package` which also contain the body.

the current implementation contains quirks to handle this approach - which results in that package body can be overwritten at any time - I think if the package already exists a `create package` call must end with an exception

Reply:
I would be happier with a single "create package" only but I don't think we can remove it since it's part of the language. I added more validation around creating the body+header.

Unless "OR REPLACE" is specified, we don't replace the package header if there is header or body or both already exists. 

If the body exits and one replaces the header (with OR REPLACE) then we create the header with a new empty body (This supposed to prevent having incompatible header + bodies).

We don't allow replacing the body if it's already exists unless OR REPLACE is specified.

Replacing the body to an existing header is allowed, it's the user's responsibility to ensure compatibility between those two. This is effectively the same as creating the header and the body later.